### PR TITLE
Fixies goreportcards

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

**Comment by nplanel** [comment]
run selenium-tests

**Comment by nplanel** [comment]
[awsome-go PR](https://github.com/avelino/awesome-go/pull/1466)

File: agent/agent.go [hunk 1]
```diff
@@ -65,6 +66,8 @@ type Agent struct {
 	TIDMapper           *topology.TIDMapper
 }
 
+// NewAnalyzerWSClientPool create a new http WebSocket client Pool
```
Comment (safchain) [inline agent/agent.go]: creates

File: agent/agent.go [hunk 2]
```diff
@@ -170,6 +175,7 @@ func (a *Agent) Stop() {
 	a.TIDMapper.Stop()
 }
 
+// NewAgent instanciate a new Agent aim to launch probes (topology and flow)
```
Comment (safchain) [inline agent/agent.go]: instanciates

File: agent/agent.go [hunk 3]
```diff
@@ -210,6 +216,8 @@ func NewAgent() *Agent {
 	}
 }
 
+// CreateRootNode create a graph.Node based on the host properties
```
Comment (safchain) [inline agent/agent.go]: CreateRootNode creates a graph.Node based on the host properties and aims to have an unique ID

File: agent/probes.go [hunk 4]
```diff
@@ -31,6 +31,7 @@ import (
 	tprobes "github.com/skydive-project/skydive/topology/probes"
 )
 
+// NewTopologyProbeBundleFromConfig create a new topology probe.ProbeBundle based on configuration
```
Comment (safchain) [inline agent/probes.go]: creates...on the configuration

File: agent/topology_forwarder.go [hunk 5]
```diff
@@ -53,6 +53,7 @@ func (t *TopologyForwarder) triggerResync() {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.SyncReplyMsgType, t.Graph))
 }
 
+// OnConnected event
```
Comment (safchain) [inline agent/topology_forwarder.go]: OnConnected websocket event handler

File: agent/topology_forwarder.go [hunk 6]
```diff
@@ -63,6 +64,7 @@ func (t *TopologyForwarder) OnConnected(c *shttp.WSAsyncClient) {
 	}
 }
 
+// OnDisconnected event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 7]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
 	}
 }
 
+// OnNodeUpdated event
```
Comment (safchain) [inline agent/topology_forwarder.go]: OnNodeUpdated graph event handler

File: agent/topology_forwarder.go [hunk 8]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnNodeUpdated(n *graph.Node) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.NodeUpdatedMsgType, n))
 }
 
+// OnNodeAdded event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 9]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnNodeAdded(n *graph.Node) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.NodeAddedMsgType, n))
 }
 
+// OnNodeDeleted event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 10]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnNodeDeleted(n *graph.Node) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.NodeDeletedMsgType, n))
 }
 
+// OnEdgeUpdated event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 11]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnEdgeUpdated(e *graph.Edge) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.EdgeUpdatedMsgType, e))
 }
 
+// OnEdgeAdded event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 12]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnEdgeAdded(e *graph.Edge) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.EdgeAddedMsgType, e))
 }
 
+// OnEdgeDeleted event
```
Comment (safchain) [inline agent/topology_forwarder.go]: ditto

File: agent/topology_forwarder.go [hunk 13]
```diff
@@ -72,30 +74,37 @@ func (t *TopologyForwarder) OnDisconnected(c *shttp.WSAsyncClient) {
...
 func (t *TopologyForwarder) OnEdgeDeleted(e *graph.Edge) {
 	t.WSAsyncClientPool.SendWSMessageToMaster(shttp.NewWSMessage(graph.Namespace, graph.EdgeDeletedMsgType, e))
 }
 
+// NewTopologyForwarder is a mechanism aim to distribute all graph node notification to WebSocket client pool.
```
Comment (safchain) [inline agent/topology_forwarder.go]: NewTopologyForwarder is a mechanism aiming to distribute all graph node notifications to a WebSocket client pool

File: agent/topology_forwarder.go [hunk 14]
```diff
@@ -109,6 +118,7 @@ func NewTopologyForwarder(host string, g *graph.Graph, wspool *shttp.WSAsyncClie
 	return t
 }
 
+// NewTopologyForwarderFromConfig create a TopologyForwarder from configuration (using host_id)
```
Comment (safchain) [inline agent/topology_forwarder.go]: just : NewTopologyForwarderFromConfig creates a TopologyForwarder from configuration

File: analyzer/flow_client.go [hunk 15]
```diff
@@ -35,12 +35,14 @@ import (
 	"github.com/skydive-project/skydive/logging"
 )
 
+// FlowClientPool describe a flow clients pool via websocket
```
Comment (safchain) [inline analyzer/flow_client.go]: just : FlowClientPool describes a flow client pool.

Websocket is only used to get connected/disconnected event being able to send flow to an active server

File: analyzer/flow_client.go [hunk 16]
```diff
@@ -35,12 +35,14 @@ import (
...
 type FlowClientPool struct {
 	sync.RWMutex
 	shttp.DefaultWSClientEventHandler
 	flowClients []*FlowClient
 }
 
+// FlowClient descibe a flow client connection
```
Comment (safchain) [inline analyzer/flow_client.go]: describes

File: analyzer/flow_client.go [hunk 17]
```diff
@@ -71,6 +73,7 @@ func (c *FlowClient) close() {
 	}
 }
 
+// SendFlow send a flow to the server
```
Comment (safchain) [inline analyzer/flow_client.go]: sends

File: analyzer/flow_client.go [hunk 18]
```diff
@@ -93,6 +96,7 @@ retry:
 	return nil
 }
 
+// SendFlows send flows to the server
```
Comment (safchain) [inline analyzer/flow_client.go]: sends

File: analyzer/flow_client.go [hunk 19]
```diff
@@ -102,12 +106,14 @@ func (c *FlowClient) SendFlows(flows []*flow.Flow) {
 	}
 }
 
+// NewFlowClient create a flow client and create a new connection to the server
```
Comment (safchain) [inline analyzer/flow_client.go]: creates

File: analyzer/flow_client.go [hunk 20]
```diff
@@ -102,12 +106,14 @@ func (c *FlowClient) SendFlows(flows []*flow.Flow) {
...
 func NewFlowClient(addr string, port int) *FlowClient {
 	FlowClient := &FlowClient{Addr: addr, Port: port}
 	FlowClient.connect()
 	return FlowClient
 }
 
+// OnConnected event
```
Comment (safchain) [inline analyzer/flow_client.go]: OnConnected websocket event handler

File: analyzer/flow_client.go [hunk 21]
```diff
@@ -124,6 +130,7 @@ func (p *FlowClientPool) OnConnected(c *shttp.WSAsyncClient) {
 	p.flowClients = append(p.flowClients, NewFlowClient(c.Addr, c.Port))
 }
 
+// OnDisconnected event
```
Comment (safchain) [inline analyzer/flow_client.go]: ditto

File: analyzer/flow_client.go [hunk 22]
```diff
@@ -137,6 +144,7 @@ func (p *FlowClientPool) OnDisconnected(c *shttp.WSAsyncClient) {
 	}
 }
 
+// SendFlows send flows using a random connection
```
Comment (safchain) [inline analyzer/flow_client.go]: sends

File: analyzer/flow_server.go [hunk 23]
```diff
@@ -45,39 +45,47 @@ import (
 	"github.com/skydive-project/skydive/topology/graph"
 )
 
+// ErrFlowUDPAcceptNotSupported error the connection can't accept as it's UDP based
 var ErrFlowUDPAcceptNotSupported = errors.New("UDP connection is datagram based (not connected), accept() not supported")
 
+// FlowConnectionType describe an UDP or TLS connection
```
Comment (safchain) [inline analyzer/flow_server.go]: describes

File: analyzer/flow_server.go [hunk 24]
```diff
@@ -45,39 +45,47 @@ import (
...
 type FlowConnectionType int
 
 const (
+	// UDP connection
 	UDP FlowConnectionType = 1 + iota
+	// TLS connection
 	TLS
 )
 
+// FlowServerConn describe a flow server connection
```
Comment (safchain) [inline analyzer/flow_server.go]: describes

going to stop here to fix the missing "s" for the third person but please recheck everything

File: api/alert.go [hunk 25]
```diff
@@ -28,7 +28,9 @@ import (
 	"github.com/nu7hatch/gouuid"
 )
 
+// Alert is a set of parameters, the Alert Action will Trigger according to his Expression.
```
Comment (safchain) [inline api/alert.go]: s/his/its/

File: common/capture.go [hunk 26]
```diff
@@ -22,12 +22,14 @@
 
 package common
 
+// CaptureType describe
```
Comment (safchain) [inline common/capture.go]: describe ??

File: common/port_allocator.go [hunk 27]
```diff
@@ -28,17 +28,21 @@ import (
...
 var (
-	InvalidPortRange = errors.New("Invalid port range")
-	NoPortLeft       = errors.New("No free port left")
+	// ErrInvalidPortRange invalid port range
+	ErrInvalidPortRange = errors.New("Invalid port range")
+	// ErrNoPortLeft no port left in the range
+	ErrNoPortLeft = errors.New("No free port left")
 )
 
+// PortAllocator describe a thread safe port list that can be allocate
```
Comment (safchain) [inline common/port_allocator.go]: allocated

File: flow/server.go [hunk 28]
```diff
@@ -29,16 +29,19 @@ import (
 	"github.com/skydive-project/skydive/logging"
 )
 
+// Namespace "Flow"
 const (
 	Namespace = "Flow"
 )
 
+// TableServer describe a mechanism to Query a flow table via flowSet in JSON
```
Comment (safchain) [inline flow/server.go]: s/via flowset in JSON/via Websocket/

File: flow/storage/elasticsearch/elasticsearch.go [hunk 29]
```diff
@@ -308,17 +312,20 @@ func (c *ElasticSearchStorage) SearchFlows(fsq filters.SearchQuery) (*flow.FlowS
 	return flowset, nil
 }
 
+// Start the DB
```
Comment (safchain) [inline flow/storage/elasticsearch/elasticsearch.go]: Start the client

File: flow/storage/elasticsearch/elasticsearch.go [hunk 30]
```diff
@@ -308,17 +312,20 @@ func (c *ElasticSearchStorage) SearchFlows(fsq filters.SearchQuery) (*flow.FlowS
...
 func (c *ElasticSearchStorage) Start() {
 	go c.client.Start([]map[string][]byte{
 		{"metric": []byte(metricMapping)},
 		{"flow": []byte(flowMapping)}},
 	)
 }
 
+// Stop the DB
```
Comment (safchain) [inline flow/storage/elasticsearch/elasticsearch.go]: ditto

File: flow/storage/orientdb/orientdb.go [hunk 31]
```diff
@@ -224,15 +228,19 @@ func (c *OrientDBStorage) SearchMetrics(fsq filters.SearchQuery, metricFilter *f
 	return metrics, nil
 }
 
+// Start the DB
```
Comment (safchain) [inline flow/storage/orientdb/orientdb.go]: s/DB/client/

File: flow/storage/orientdb/orientdb.go [hunk 32]
```diff
@@ -224,15 +228,19 @@ func (c *OrientDBStorage) SearchMetrics(fsq filters.SearchQuery, metricFilter *f
...
 func (c *OrientDBStorage) Start() {
 }
 
+// Stop the DB
```
Comment (safchain) [inline flow/storage/orientdb/orientdb.go]: ditto

File: flow/storage/orientdb/orientdb.go [hunk 33]
```diff
@@ -224,15 +228,19 @@ func (c *OrientDBStorage) SearchMetrics(fsq filters.SearchQuery, metricFilter *f
...
 func (c *OrientDBStorage) Stop() {
 }
 
+// Close the DB
```
Comment (safchain) [inline flow/storage/orientdb/orientdb.go]: ditto

File: flow/table.go [hunk 34]
```diff
@@ -48,22 +48,26 @@ type TableReply struct {
...
-type FlowHandler struct {
+// Handler define a flow callback called every time
+type Handler struct {
 	callback ExpireUpdateFunc
 	every    time.Duration
 }
 
-func NewFlowHandler(callback ExpireUpdateFunc, every time.Duration) *FlowHandler {
-	return &FlowHandler{
+// NewFlowHandler create a flow callback handler that would be asynchronously called every time
```
Comment (safchain) [inline flow/table.go]: s/would/will/

File: ovs/ovsdb.go [hunk 35]
```diff
@@ -315,13 +326,15 @@ func (o *OvsMonitor) setMonitorRequests(table string, r *map[string]libovsdb.Mon
...
 
+// AddMonitorHandler subscribe a new monitor events handler
 func (o *OvsMonitor) AddMonitorHandler(handler OvsMonitorHandler) {
 	o.Lock()
 	defer o.Unlock()
 
 	o.MonitorHandlers = append(o.MonitorHandlers, handler)
 }
 
+// ExcludeColumn exclude some column to be montiored
```
Comment (safchain) [inline ovs/ovsdb.go]: monitored

File: storage/orientdb/client.go [hunk 36]
```diff
@@ -265,6 +276,7 @@ func NewClient(url string, database string, username string, password string) (*
 	return client, nil
 }
 
+// Request send a reques to the OrientDB server
```
Comment (safchain) [inline storage/orientdb/client.go]: request

File: storage/orientdb/client.go [hunk 37]
```diff
@@ -374,6 +390,7 @@ func (c *Client) Upsert(doc Document, key string) (Document, error) {
 	return nil, err
 }
 
+// GetDocumentClass return an OrientDB docuemnt class
```
Comment (safchain) [inline storage/orientdb/client.go]: document

File: topology/graph/graph.go [hunk 38]
```diff
@@ -87,16 +91,19 @@ type graphElement struct {
 	revision  int64
 }
 
+// Node of the graph
 type Node struct {
 	graphElement
 }
 
+// Edge of the graph identified by a parent and a child
```
Comment (safchain) [inline topology/graph/graph.go]: not identified but linking

File: topology/graph/graph.go [hunk 39]
```diff
@@ -116,10 +123,13 @@ type GraphBackend interface {
 	WithContext(graph *Graph, context GraphContext) (*Graph, error)
 }
 
+// GraphContext describe within time slice
 type GraphContext struct {
 	TimeSlice *common.TimeSlice
 }
 
+// Graph describe the graph object based on events and context mechanism
+// An associate backend is used as storage
```
Comment (safchain) [inline topology/graph/graph.go]: associated

File: topology/graph/metrics.go [hunk 40]
```diff
@@ -102,6 +104,7 @@ func (im *InterfaceMetric) GetFieldInt64(field string) (int64, error) {
 	return 0, common.ErrFieldNotFound
 }
 
+// Add do a sum opreation on interface metric
```
Comment (safchain) [inline topology/graph/metrics.go]: operation

File: topology/graph/orientdb.go [hunk 41]
```diff
@@ -212,6 +221,7 @@ func (o *OrientDBBackend) GetEdge(i Identifier, t *common.TimeSlice) (edges []*E
 	return
 }
 
+// GetEdgeNodes return the parents and child nodes of an edge within time slice, matching metadatas
```
Comment (safchain) [inline topology/graph/orientdb.go]: metadata without "s"

File: topology/graph/traversal/traversal.go [hunk 42]
```diff
@@ -1176,6 +1250,40 @@ nodeloop:
 	return ntv
 }
 
+// InE step of an edge
```
Comment (safchain) [inline topology/graph/traversal/traversal.go]: of node not edge and why moving it ?

**Comment by nplanel** [comment]
done

**Comment by nplanel** [comment]
rerun tests

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by safchain** [comment]
I made a PR on your branch fixiing typos. Moreover we can merge your patch as it is, go-fmt is failing

**Pull request merged**